Marlin/gcode.h

```diff
@@ -139,6 +149,17 @@ class GCodeParser {
       return b;
     }
 
+    static volatile bool seen_any() {
+      return strlen(command_args)!=0; // todo: is this right?
+    }
+
+    static volatile bool seen_any_axis() {
+      return !!strchr(command_args, 'X') || //X
+             !!strchr(command_args, 'Y') || //Y
+             !!strchr(command_args, 'Z') || //Z
+             !!strchr(command_args, 'E'); //E
+    }
```

Comment:
For concision and to make the meaning more clear…
```cpp
#define SEEN_TEST(L) !!strchr(command_args, L)
return SEEN_TEST('X') || SEEN_TEST('Y') || SEEN_TEST('Z') || SEEN_TEST('E');
```
Since the `seen_any_axis` function will be the same in both cases, it can be defined just once, below.